### PR TITLE
Filter out language-specific topic sections when creating See Also sections

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: rdar://137311688 

## Summary

This fixes a bug where language specific authored topic sections was considered when creating See Also sections for all language representations.

There's still a possible issue where a symbol with different canonical containers in different languages would pick the wrong container and not find the most appropriate topic section, but that requires some utilities from either #1081 or #1075 to fix, so I'm deferring that.

## Dependencies

None

## Testing

- Recreate [the setup from this new test](https://github.com/swiftlang/swift-docc/pull/1082/files#diff-2f4773195a122c13ab57aa39efac9235a4c402cc5b0ab1037ee97a4515f88501R3353-R3383) with a few (at least 3) symbols that exist in more than one language and two language specific topic sections that each curate different subsets of those symbols.

- Then preview the documentation for one of the symbols that is curated into both language specific topic sections. 
  - On the Swift version of the page the See Also section at the bottom should display the title and links from the Swift-only topic section
  - On the Objective-C version of the page the See Also section at the bottom should display the title and links from the Objective-C-only topic section

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- ~[ ] Updated documentation if necessary~ Not applicable

### CODE REVIEW

File: Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift
Comment (patshaughnessy) [inline Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift:166]: Do we plan on addressing this soon? Should we open a GitHub issue?

File: Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift
Comment (patshaughnessy) [inline Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift:198]: Is this just refactoring what we had earlier? Would it be even simpler to intersect the arrays somehow, vs. a nested loop? Not really a concern since the number of languages is usually 1 or 2. Would just be more readable.

Maybe something like this: 


```
let variantLanguages = Set(variantsTraits.compactMap{ $0.interfaceLanguage }.map{ SourceLanguage(id: $0)})

etc...


// Filter out nodes that aren't available in any of the given traits.
!context.sourceLanguages(for: reference).intersection(variantLanguages).isEmpty

```


Although you use `variantLanguageIDs` above in `isRelevant` so we'd need some more refactoring.

**Review by patshaughnessy — APPROVED** [review]
Looks good 👍 I was able to observe the broken behavior on the main branch, and the fix in action on this branch.

Left a nerd snipe code suggestion :)

File: Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift
Comment (d-ronnqvist) [reply Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift:198]: Yes, I noticed that the two separate `filters` could be expressed as once combined condition and since `isRelevant` also does a language check I wanted both places to do the same check (and felt that it was nice to get change one of the `contains(where:)` to a simpler `contains(_:)`.

Because the languages is going to be incredibly small (either 1 or 2) any performance difference is not going to be noticeable so I can update it to use `Set<SourceLanguage>` in both places for readability.

File: Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift
Comment (d-ronnqvist) [reply Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift:166]: Yes, I started working on this issue but it requires some replacement API that I introduced in a related PR so I put it on hold until those are merged. I opened [this issue](https://github.com/swiftlang/swift-docc/issues/1085) that refers to this code. I'll add a link in the code comment as well

File: Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift
Comment (d-ronnqvist) [reply Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift]: I updated to use a set of languages in https://github.com/swiftlang/swift-docc/pull/1082/commits/693910913d772eeecff239022f554ac48130d720.

**Review by franklinsch — APPROVED** [review]

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**
